zobrist/hash.go

```diff
@@ -1,43 +1,60 @@
 package zobrist
 
-import "lukechampine.com/frand"
+import (
+	"lukechampine.com/frand"
 
-const bignum = 1 << 63
+	"github.com/domino14/macondo/alphabet"
+)
+
+const bignum = 1 << 63 - 1
 
 // generate a zobrist hash for a crossword game position.
 // https://en.wikipedia.org/wiki/Zobrist_hashing
 type Zobrist struct {
-	p2ToMove int64
+	p2ToMove uint64
 	// note that the endgame only cares about 2 zeros in a row.
 	// if we use this hash for non-endgame positions we might
 	// want to turn this into an array as well, with each element
 	// corresponding to the number of zero positions.
-	lastMoveWasZero int64
+	lastMoveWasZero uint64
 
-	posTable  [][]int64
-	rackTable [][]int64
+	posTable  [][]uint64
+	rackTable [][]uint64
 }
 
 func (z *Zobrist) Initialize(boardDim int, numtiles int, numblankletters int) {
 
-	z.posTable = make([][]int64, boardDim*boardDim)
+	z.posTable = make([][]uint64, boardDim*boardDim)
 	for i := 0; i < boardDim*boardDim; i++ {
 		// for example, 27 types of tiles + 26 letters the blank can be
 		// notice that in norwegian, the blank can be a tile that's not
 		// one of the tiles in the distribution.
-		z.posTable[i] = make([]int64, numtiles+numblankletters)
-		for j := 0; j < numtiles+numblankletters; j++ {
-			z.posTable[i][j] = int64(frand.Uint64n(bignum))
+		z.posTable[i] = make([]uint64, 256)
```

Comment:
what happens if you use `numtiles+numblankletters`?

Reply:
I was observing index out of bounds accesses... spent the better part of an hour trying to make it work before trying `256` instead.

Reply:
I've committed this suggestion.